### PR TITLE
add SOLOv2 model

### PR DESCRIPTION
**Pull request by still-wait**
as title.

### CODE REVIEW

File: ppdet/modeling/anchor_heads/solov2_head.py [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+    def get_loss(self,
+                 cate_preds,
+                 kernel_preds,
+                 ins_pred,
+                 ins_labels,
+                 cate_labels,
+                 grid_order_list,
+                 fg_num,
+                 grid_offset,
+                 batch_size=1):
```
Comment (qingqing01) [inline ppdet/modeling/anchor_heads/solov2_head.py:247]: 同上，YOLOv3的loss里依赖batch-size不够友好

File: ppdet/modeling/architectures/solov2.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    __category__ = 'architecture'
+    __inject__ = ['backbone', 'fpn', 'bbox_head', 'mask_head']
+
+    def __init__(self,
+                 backbone,
+                 fpn=None,
+                 bbox_head='SOLOv2Head',
+                 mask_head='SOLOv2MaskHead',
+                 batch_size=1):
```
Comment (qingqing01) [inline ppdet/modeling/architectures/solov2.py:50]: 同上，网络中尽量依赖输入获取batch-size，而不是外部传入

File: ppdet/modeling/mask_head/__init__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+from . import solo_mask_head
+
+from .solo_mask_head import *
```
Comment (qingqing01) [inline ppdet/modeling/mask_head/__init__.py:19]: 安装下pre-commit吧

File: deploy/python/infer.py [hunk 4]
```diff
@@ -67,7 +67,8 @@ def __init__(self,
                  max_size,
                  use_cv2=True,
                  image_shape=None,
-                 interp=cv2.INTER_LINEAR):
+                 interp=cv2.INTER_LINEAR,
+                 resize_bbox=False):
```
[the remark this answers is not in the document]
Comment (still-wait) [reply deploy/python/infer.py:71]: When export model, `resize_bbox` will be load in `infer_cfg.yml`, if not add it, will error.

File: ppdet/modeling/anchor_heads/solov2_head.py (hunk 1 above)
Comment (still-wait) [reply ppdet/modeling/anchor_heads/solov2_head.py:248]: for循环里需要常量，计算loss也是对每个img计算，暂时没想到更好的解决方案。

File: ppdet/modeling/architectures/solov2.py (hunk 2 above)
Comment (still-wait) [reply ppdet/modeling/architectures/solov2.py:50]: Same as above.

File: ppdet/data/transform/batch_operators.py [hunk 5]
```diff
@@ -590,3 +599,147 @@ def draw_truncate_gaussian(self, heatmap, center, h_radius, w_radius):
...
+        self.num_grids = num_grids
+        self.scale_ranges = scale_ranges
+        self.coord_sigma = coord_sigma
+        self.sampling_ratio = sampling_ratio
+
+    def _scale_size(self, im, scale):
+        h, w = im.shape[:2]
+        new_size = (int(w * float(scale) + 0.5), int(h * float(scale) + 0.5))
+        #resized_img = cv2.resize(
+        #        im, new_size, dst=None, interpolation=cv2.INTER_LINEAR)
```
[the remark this answers is not in the document]
Comment (still-wait) [reply ppdet/data/transform/batch_operators.py]: Done, thanks.

File: ppdet/modeling/anchor_heads/solov2_head.py [hunk 6]
```diff
@@ -0,0 +1,531 @@
...
+        # CoordConv
+        x_range = paddle.linspace(
+            -1, 1, fluid.layers.shape(ins_kernel_feat)[-1], dtype='float32')
+        y_range = paddle.linspace(
+            -1, 1, fluid.layers.shape(ins_kernel_feat)[-2], dtype='float32')
+        y, x = paddle.tensor.meshgrid([y_range, x_range])
+        x = fluid.layers.unsqueeze(x, [0, 1])
+        y = fluid.layers.unsqueeze(y, [0, 1])
+        y = fluid.layers.expand(y, expand_times=[batch_size, 1, 1, 1])
+        x = fluid.layers.expand(x, expand_times=[batch_size, 1, 1, 1])
```
[the remark this answers is not in the document]
Comment (still-wait) [reply ppdet/modeling/anchor_heads/solov2_head.py]: Done, thanks.

File: ppdet/modeling/mask_head/__init__.py (hunk 3 above)
Comment (still-wait) [reply ppdet/modeling/mask_head/__init__.py]: Already executed `pre-commit`

**Pull request closed without merging**